Source/WebCore/rendering/TextPaintStyle.cpp

```diff
@@ -148,6 +148,9 @@ TextPaintStyle computeTextSelectionPaintStyle(const TextPaintStyle& textPaintSty
     if (emphasisMarkForeground.isValid() && emphasisMarkForeground != selectionPaintStyle.emphasisMarkColor)
         selectionPaintStyle.emphasisMarkColor = emphasisMarkForeground;
 
+    if (auto renderStyle = renderer.parent()->selectionPseudoStyle())
```

Comment:
Why is this using `renderer.parent()`, and not `renderer`?

Reply:
`MarkedText::Type::Highlight` takes styles based on `if (auto renderStyle = renderer.parent()->getUncachedPseudoStyle({ PseudoId::Highlight, markedText.highlightName }, &renderer.style()))` so I assumed that it would make sense for selection to follow the same procedure. I'll look into why though.